File: 분류 대기/11047.py
# 11047번 동전 0

n, k = map(int, input().split())
count = 0
coins = []
for _ in range(n):
    coins.append(int(input()))

coins = sorted(coins, reverse=True)

for coin in coins:
    count += k // coin
    k %= coin

print(count)

File: 분류 대기/11047.cpp
// 11047번 동전 0

#include <iostream>

using namespace std;

int main() {
    int n, k, count = 0;
    int coins[10] = {};

    cin >> n >> k;

    for (int i = n - 1; i >= 0; --i) {
        cin >> coins[i];
    }

    for (int i = 0; i < n; ++i) {
        count += k / coins[i];
        k %= coins[i];
    }

    cout << count << endl;
    return 0;
}